### ISSUE TITLE
Missing mate-in-one when depth high

### ISSUE THREAD

**Issue by znxftw** [znxftw/rudim#49]
Noticed this in one of the games Rudim played on lichess with another bot.
Rudim won, but seems to be missing mate in one in this position as soon as `depth >= 4`
FEN : `8/5pkp/6p1/8/1q5P/8/K7/4q3 b - - 0 52`

```
uci
id name Rudim 1.4
id author Vishnu B
uciok
position fen 8/5pkp/6p1/8/1q5P/8/K7/4q3 b - - 0 52
go depth 1
info depth 1 score cp 2287 nodes 52 time 1 nps 52000
bestmove b4h4
go depth 2
info depth 1 score cp 2287 nodes 0 time 1 nps 0
info depth 2 score cp 48999 nodes 170 time 1 nps 170000
bestmove e1b1
go depth 3
info depth 1 score cp 48999 nodes 0 time 1 nps 0
info depth 2 score cp 48999 nodes 0 time 1 nps 0
info depth 3 score cp 48999 nodes 2748 time 2 nps 1374000
bestmove e1b1
go depth 4
info depth 1 score cp 48999 nodes 0 time 1 nps 0
info depth 2 score cp 48999 nodes 0 time 1 nps 0
info depth 3 score cp 48999 nodes 0 time 1 nps 0
info depth 4 score cp 48999 nodes 5806 time 15 nps 387066
bestmove b4b1
```

**Comment by znxftw** [comment]
b4b1 should ideally have cp = 48997 mate in 3 plies, not 1.

**Comment by znxftw** [comment]
Fixed in 447c5f241976d384d803dbf397cbdcd78b534e02
